lib/utils/system_tray.dart

```diff
@@ -23,79 +23,89 @@ class DesktopSystemTray extends GetxService {
-            if (playerController.currentQueue.isNotEmpty) {
-              playerController.playPause();
-            }
-          }),
-      MenuItemLabel(
-          label: 'Next',
-          onClicked: (menuItem) {
-            if (playerController.currentQueue.isNotEmpty) {
-              playerController.next();
-            }
-          }),
-      MenuSeparator(),
-      MenuItemLabel(
-          label: 'Quit',
-          onClicked: (menuItem) async {
-            await Get.find<AudioHandler>().customAction("saveSession");
-            exit(0);
-          }),
+    final Menu menu = Menu(items: [
+      MenuItem(
+        label: 'Show/Hide',
+        onClick: (menuItem) async => await windowManager.isVisible()
+            ? await windowManager.hide()
+            : await windowManager.show(),
+      ),
+      MenuItem.separator(),
+      MenuItem(
+        label: 'Prev',
+        onClick: (menuItem) {
+          if (playerController.currentQueue.isNotEmpty) {
+            playerController.prev();
+          }
+        },
+      ),
+      MenuItem(
+        label: 'Play/Pause',
+        onClick: (menuItem) {
+          if (playerController.currentQueue.isNotEmpty) {
+            playerController.playPause();
+          }
+        },
+      ),
+      MenuItem(
+        label: 'Next',
+        onClick: (menuItem) {
+          if (playerController.currentQueue.isNotEmpty) {
+            playerController.next();
+          }
+        },
+      ),
+      MenuItem.separator(),
+      MenuItem(
+        label: 'Quit',
+        onClick: (menuItem) async {
+          await Get.find<AudioHandler>().customAction("saveSession");
+          exit(0);
+        },
+      ),
     ]);
 
     // set context menu
-    await systemTray.setContextMenu(menu);
+    await trayManager.setContextMenu(menu);
 
-    // handle system tray event
-    systemTray.registerSystemTrayEventHandler((eventName) {
-      if (eventName == kSystemTrayEventClick) {
-        GetPlatform.isWindows
-            ? windowManager.show()
-            : systemTray.popUpContextMenu();
-      } else if (eventName == kSystemTrayEventRightClick) {
-        GetPlatform.isWindows
-            ? systemTray.popUpContextMenu()
-            : windowManager.show();
-      }
-    });
     await windowManager.setPreventClose(true);
     listener = CloseWindowListener();
     windowManager.addListener(listener);
   }
 
   @override
   void onClose() {
+    trayManager.removeListener(this);
     windowManager.removeListener(listener);
     super.onClose();
   }
+
+  @override
+  void onTrayIconMouseDown() {
+    if (GetPlatform.isWindows) {
+      windowManager.show();
+    } else {
+      trayManager.popUpContextMenu();
+    }
+
+    super.onTrayIconMouseDown();
+  }
+
+  @override
+  void onTrayIconRightMouseDown() {
+    if (GetPlatform.isWindows) {
+      trayManager.popUpContextMenu();
+    } else {
+      windowManager.show();
+    }
+
+    super.onTrayIconRightMouseDown();
+  }
```

Comment:
I'm not certain this is the click behaviour we want, I simply reproduced what we currently had.

Personally, I think left-click to show/hide and right-click for context menu sounds quite appropriate and expected (speaking as a full-time Linux user).